gitflow/readme.md

```diff
@@ -0,0 +1,318 @@
+* code review - pull request allows to increase code quality and knowledge rotation in team, and as a consequence reduce "bus factor"
+
+#### Cons:
+* "scary merge" - well illustrated problem by Fowler, to avoid it keep semantic changes as separate branches(explained later)
+* "afraid of refactoring" - semantic changes(refacroring, renaming) leads to "scary merges", so try to do it when code targeted by refactoring is not changed by any other branches
+* dependencies(libraries) update - makes Pull Requests view polluted by library files and in general useless, so keep this updates as a separate branches
+
+##Table of contents
+1. Required reading
+2. Branches
+3. Environments
+4. Branch name conventions
+5. Pull Requests
+6. Commit and Pull Requests message guideline
+7. Initiate repository with main branches
+8. Feature branches flow
+9. Hotfix branches flow
+10. Release branches flow
+
+## Required readings
+
+Take attention to Feature Branching with Pull Requests
+
+1. [Try github](https://try.github.io/levels/1/challenges/1)
+2. [Atlassian git tutorial](https://www.atlassian.com/git/tutorials/setting-up-a-repository/ ) (except Migrate to Git from SVN)
+3. [Feature branching classic by Martin Fowler](http://martinfowler.com/bliki/FeatureBranch.html)
+4. [Closest to ours branching model by Vincent Driessen](http://nvie.com/posts/a-successful-git-branching-model/)
+5. [Vincent Driessen git-flow cheatsheet](http://danielkummer.github.io/git-flow-cheatsheet/)
+([ru](http://danielkummer.github.io/git-flow-cheatsheet/index.ru_RU.html))
+
+### Supporting readings
+
+1. [Semantic Versioning 2.0.0](http://semver.org/)
+2. [Git cheat sheet](https://training.github.com/kit/downloads/github-git-cheat-sheet.pdf)
+3. [Github help: user collaborating](https://help.github.com/categories/collaborating/)
+
+## Branches
+
+- `master` - always **stable** and **release ready** branch
+- `development` - default branch, contains latest **features** and **fixes**
+- `feature-*` - branches for feature development and decencies update
+- `realese-*` - branches for release(production) version
+- `hotfix-*` - branches for release `fixes`
+- `refactor-*` - branches for semantic changes
+
+## Environments
+
+- `master` - production
+- `development` - staging
+- `release-*` - for history reasons or supporting several versions at production
+- `feature-*` - continuous deployment to test environment
+- `hotfix-*` - development (local)
+
+
+## Branch name conventions
+
+Branch names should contain only `lowercase letters`, `numbers` and `hyphens`,
+regexp: `/[a-z0-9\-]+/`
+
+**Important** : why not with slashes, like in git flow by Vincent Driessen? -
+ because of continuous deployment to corresponding subdomain,
+ for example: `feature-auth` will be deployed to `auth.project.com`
+
+## Pull Requests
+
+* For each pull request it should be an issue (except documentation)
+* Pull Request must have a description, which should explain what was done and how it affects project
```

Comment:
@korel-san task should say: what you should achieve
and PR description: what you did